R/LSproblemofpointscommon.R

```diff
@@ -1,227 +1,234 @@
-compare_function1 <- function(p,m1,n1,t,simulation){
+compareChanceTwoPlayers <- function(p,m1,n1,t,simulation){
 
   # first estimating the probability that player 1 wins by simulating the game for a given number of times.
   m <- t - m1 # m: the number of success that player 1 should have to win the game
   n <- t - n1 # n: the number of success that player 2 should have to win the game
 
-  record_game <- vector()
+  recordGame <- vector()
   for (i in 1:simulation){
     # copy m and n
-    record_trial <- rbind(m, n)
+    recordTrial <- rbind(m, n)
     # keep running the game until one of them wins
-    while (record_trial[1, ] * record_trial[2, ] != 0){
-      record_trial <- record_trial - rmultinom(1, 1, c(p, 1 - p)) #updating the recorded result
+    while (recordTrial[1, ] * recordTrial[2, ] != 0){
+      recordTrial <- recordTrial - rmultinom(1, 1, c(p, 1 - p)) #updating the recorded result
     }
 
-    # if record_trial[1,] = 0, record 1,player 1 wins this single trial;
-    # if record_trial[2,] = 0, record 0, player 2 wins
-    record_game[i] <- record_trial[2, ]^record_trial[1, ]
+    # if recordTrial[1, ] = 0, record 1,player 1 wins this single trial;
+    # if recordTrial[2, ] = 0, record 0, player 2 wins
+    recordGame[i] <- recordTrial[2, ]^recordTrial[1, ]
 
   }
-  p_simulated <- sum(record_game)/simulation    # estimated probability that player 1 wins
+  pSimulated <- sum(recordGame)/simulation    # estimated probability that player 1 wins
 
   # plot the simulated probability over the number of times the simulation is performed
-  p_cumulative <- record_game
+  pCumulative <- recordGame
   for (i in 2:simulation){
-    p_cumulative[i] <- (p_cumulative[i] + p_cumulative[i-1]*(i-1))/i
+    pCumulative[i] <- (pCumulative[i] + pCumulative[i-1]*(i-1))/i
   }
-  #plot(p_cumulative,
-  #main="Simulated probability of player 1 winning over simulations",
-  #xlab = "Number of simulations",ylab="Probability of winning",type="l",
-  #xlim = c(0, simulation), ylim = c(0, 1),)
 
   # calculating the probabiltiy that player 1 wins from negative binomial distribution
-  p_calculated <- pnbinom(n-1,m,p)
+  pCalculated <- pnbinom(n-1,m,p)
 
   # difference between simulated and calculated probability
-  dif = abs(p_simulated - p_calculated)
-  return (list(p_simulated,p_calculated,dif,p_cumulative))
+  dif = abs(pSimulated - pCalculated)
+  return (list(pSimulated,pCalculated,dif,pCumulative))
 }
 
-compare_function2 <- function(k1, t, p, simulation){
+compareChanceNPlayers <- function(k1, t, p, simulation){
   # first estimating the probability that player 1 wins
   k <- t-k1 # k: the vector regarding the number of more successes that player 1, 2, ..., n should have to win the game
 
   # normalize the probability if they do not sum to one
   p <- p/sum(p)
 
-  record_game <- vector()# initializing the vector to record the results of all the games
+  recordGame <- vector()# initializing the vector to record the results of all the games
```

Comment:
preallocating the data is typically a good idea
```suggestion
  recordGame <- numeric(simulation)# initializing the vector to record the results of all the games
```

Reply:
I also made the same changes for the other two functions.